camel/messages/base.py

```diff
@@ -26,370 +26,306 @@
     OpenAIUserMessage,
 )
 from camel.prompts import CodePrompt, TextPrompt
-from camel.types import (
-    OpenAIBackendRole,
-    OpenAIImageType,
-    OpenAIVisionDetailType,
-    RoleType,
-)
+from camel.types import MessageType, OpenAIBackendRole, RoleType
 from camel.utils import Constants
 
 
-@dataclass
-class BaseMessage:
-    r"""Base class for message objects used in CAMEL chat system.
+class Sender(BaseModel):
+    id: str
+
+
+class Receiver(BaseModel):
+    id: str
+
 
-    Args:
-        role_name (str): The name of the user or assistant role.
-        role_type (RoleType): The type of role, either :obj:`RoleType.
-            ASSISTANT` or :obj:`RoleType.USER`.
-        meta_dict (Optional[Dict[str, str]]): Additional metadata dictionary
-            for the message.
-        content (str): The content of the message.
-        video_bytes (Optional[bytes]): Optional bytes of a video associated
-            with the message. Default is None.
-        image_list (Optional[List[Image.Image]]): Optional list of PIL Image
-            objects associated with the message. Default is None.
-        image_detail (Literal["auto", "low", "high"]): Detail level of the
-            images associated with the message. Default is "auto".
-        video_detail (Literal["auto", "low", "high"]): Detail level of the
-            videos associated with the message. Default is "low".
-    """
+class Content(BaseModel):
+    text: Optional[List[str]] = Field(default_factory=list)
+    image_url: Optional[List[str]] = Field(default_factory=list)
+    image_detail: str = Field(default="auto")
+    video_url: Optional[List[str]] = Field(default_factory=list)
+    video_detail: str = Field(default="low")
+    audio_url: Optional[List[str]] = Field(default_factory=list)
 
+
+class Envelope(BaseModel):
+    time_sent: str
+    time_received: str
+
+
+class BaseMessage(BaseModel):
     role_name: str
     role_type: RoleType
-    meta_dict: Optional[Dict[str, str]]
-    content: str
-    video_bytes: Optional[bytes] = None
-    image_list: Optional[List[Image.Image]] = None
-    image_detail: Literal["auto", "low", "high"] = "auto"
-    video_detail: Literal["auto", "low", "high"] = "low"
+    content: Content
+    sender: Optional[Sender] = None
+    receiver: Optional[Union[Receiver, Tuple[Receiver, ...]]] = None
+    meta_dict: Optional[Dict[str, str]] = None
+    message_type: MessageType = MessageType.DEFAULT
+    subject: Optional[str] = None
+    reply_with: Optional["BaseMessage"] = None
+    in_reply_to: Optional["BaseMessage"] = None
```

Comment:
I remember that you mentioned all the messages are stored in the memory so that every time the chat agent chat, we update the memory. If all the messages are stored in a central place, maybe we could just store the message id here instead of the entire message.